[Ty-stan0417] iP

Ty-stan0417 opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Ty-stan0417 committed:
Level-0

Ty-stan0417 committed:
Level-0

Ty-stan0417 committed:
Level-1

Ty-stan0417 committed:
Level-2

Ty-stan0417 committed:
Level-3

Ty-stan0417 committed:
Level-4

Ty-stan0417 committed:
Level-4

Ty-stan0417 committed:
A-TextUiTesting

Ty-stan0417 committed:
Level-5

Ty-stan0417 committed:
Level-6

Ty-stan0417 committed:
Level-7

Ty-stan0417 committed:
Level-8

tanjieling0 commented on src/main/java/MarkNothingException.java:
```diff
@@ -0,0 +1,5 @@
+public class MarkNothingException extends Exception {
```
Perhaps there could be a better name for this exception? perhaps InvalidTaskException or something 😬

Ty-stan0417 replied:
It's a good suggestion. I already name it InvalidTaskException. Thanks.

tanjieling0 commented on src/main/java/Tyler.java:
```diff
@@ -0,0 +1,189 @@
+                          throw new EmptyNameException();
+                      }
+                      String[] rest1 = input.split(" ", 2);
+                      String[] rest2 = rest1[1].split(" /by ");
```
Could there be a better way to differentiate the variables instead of 1 and 2? The same naming convention can be seen in other parts of the code as well.

Ty-stan0417 replied:
Good suggestion. I have change the naming so that can understand what the code is through the name.

tanjieling0 commented on src/main/java/Tyler.java:
```diff
@@ -0,0 +1,189 @@
+    public static String toFileString(Task task) {
+        String name = task.name;
+        boolean isDone = task.isDone;
+        int isDoned = isDone ? 1 : 0;
```
Maybe the "is" variable naming should be reserved for boolean values whereas in this case it is an integer

Ty-stan0417 replied:
Good suggestion. I will change it later.

tanjieling0 commented on src/main/java/EmptyNameException.java:
```diff
@@ -0,0 +1,5 @@
+public class EmptyNameException extends Exception {
```
Nice naming :)

tanjieling0 reviewed (commented):
LGTM, code is fairly neat and tidy, but perhaps renaming some of the variables / classes can help make the code more readable for other code reviewers :)

Ty-stan0417 committed:
A-MoreOOP

Ty-stan0417 committed:
Organize to Packages
main package: tyler
other package: tyler.command, tyler.parser, tyler.task, tyler.exception, tyler.storage, tyler.ui

Ty-stan0417 commented:
About this A-Packages, I don't know why the Tyler directory and Command directory start with capital letter. Actually in my code, it is small letter, all my package is in small letter.

Ty-stan0417 committed:
Renaming the folder

Ty-stan0417 committed:
Rename

Ty-stan0417 committed:
Merge remote-tracking branch 'origin/heads/A-Packages'

Ty-stan0417 committed:
Merge branch 'master' into add-gradle-support

Ty-stan0417 committed:
Update gradle file

FangRuoqing reviewed (commented):
Your choice of names throughout the codebase is consistent and easy to understand, which is awesome for anyone reading your code. Maybe adding some comments for your methods and classes would be more helpful? Generally, I like how your code is nicely organized into packages, making it a breeze to find what you need.

redcolorbicycle commented on src/main/java/tyler/command/EventCommand.java:
```diff
@@ -0,0 +1,26 @@
+        int num = tasks.getNumOfTasks();
+        ui.showTaskAdded(newTask, num);
+    }
+}
```
variable names are aptly named and logic is easy to follow.

redcolorbicycle commented on src/main/java/tyler/parser/Parser.java:
```diff
@@ -0,0 +1,64 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Parser {
```
Perhaps you could consider splitting up the function - would grouping the code into a function from to be handled by another class make it a little less lengthy?

redcolorbicycle commented on src/main/java/tyler/parser/Parser.java:
```diff
@@ -0,0 +1,64 @@
+            String name = info[0];
+            LocalDateTime end = LocalDateTime.parse(info[1], INPUT_DATE_FORMAT);
+            return new DeadlineCommand(name, end);
+        } else if (command.equals("event")) {
```
Instead, would 
```
} else if (command.equals("event")) {
HandleCommand.HandleEvent;
}
```
abstract away these details and make it easier to reference? 

I like that the constants are in SCREAMING_SNAKE_CASE, classes in PascalCase and variables in camelCase!

redcolorbicycle commented on src/main/java/tyler/storage/Storage.java:
```diff
@@ -0,0 +1,71 @@
+        this.filePath = filePath;
+    }
+
+    public ArrayList<Task> load() throws IOException {
```
Similar to above, the code logic is sound and the naming convention is well done and easy to follow! Could abstracting some details into function make it even better?

redcolorbicycle commented on src/main/java/tyler/ui/Ui.java:
```diff
@@ -0,0 +1,57 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+
+public class Ui {
```
Code follows the coding standard well!

redcolorbicycle reviewed (commented):
Overall, code is named according to coding standard, is easy to understand and well written!  Some cosmetic suggestions to make reading it a little easier have been requested.
I think the code is well written! :+1:

Ty-stan0417 committed:
Adding DeleteCommandTest, ParserTest, EventTest

Ty-stan0417 committed:
Creating JAR File

Ty-stan0417 committed:
Writing JavaDoc to the class and the method.

Ty-stan0417 committed:
Tweaking the code to comply with coding standard.

Ty-stan0417 committed:
Give user a way to find a task by searching for a keyword and printing the list of matched tasks.

Ty-stan0417 committed:
Merge branch 'branch-A-CodingStandard'

Ty-stan0417 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/tyler/parser/Parser.java

Ty-stan0417 committed:
Update the description of iP using GFMD

Ty-stan0417 committed:
Using checkStyle to detect coding style violations

Ty-stan0417 committed:
Update the GFMD

Ty-stan0417 committed:
Create the GUI for the chatbot

Ty-stan0417 committed:
Add assert feature to some parts of the code.

Ty-stan0417 committed:
Improve code quality

Unify variations of execute() methods

Unifying all those duplicated code improves the code quality.

As a step towards such unification, let's extract those duplicated code
blocks into separate methods in their respective classes. Doing so will
make the subsequent unification easier.

Ty-stan0417 committed:
Merge pull request #2 from Ty-stan0417/heads/A-Assertions

Add assert feature to some parts of the code.

Ty-stan0417 committed:
Merge pull request #3 from Ty-stan0417/heads/A-CodeQuality

Improve code quality

Ty-stan0417 committed:
Provide in-App guidance to users.

Add InvalidDateTimeFormatException which throws error when the date
format is not correct.

Add WrongInputFormatException. Throws error when deadline and event
command does not follow the format.

Ty-stan0417 committed:
Add a User Guide to the project

Update the given docs\README.md

Provide the reader with enough guidance to be able to use all important
features of my chatbot.

Ty-stan0417 committed:
Update the path of Ui.png

Ty-stan0417 committed:
Update the path of Ui.png
